src/registries/npm-registry.js

```diff
@@ -139,6 +140,15 @@ export default class NpmRegistry extends Registry {
   }
 
   getRegistry(packageName: string): string {
+    // If this is a request for a file, return the registry from the URL
+    if (HTTPS_EXPRESSION.test(packageName)) {
+      const registryProtocolHost = packageName.match(/^https?:\/\/[^\/]+\//i);
```

Comment:
This only matches on the protocol + hostname. An npm registry URL can also include a path, e.g. for registries that are part of a larger artifact hosting application, like Nexus or Artifactory.